monai/data/utils.py

```diff
@@ -287,13 +288,15 @@ def iter_patch(
     padded = bool(mode)
     # pad image by maximum values needed to ensure patches are taken from inside an image
     if padded:
-        arrpad = np.pad(arr, tuple((p, p) for p in patch_size_), look_up_option(mode, NumpyPadMode).value, **pad_opts)
+        is_v = [not bool(p) for p in ensure_tuple_size(patch_size, arr.ndim)]  # no padding for (0 or None) patch_size
+        _pad_size = tuple(p if is_v else 0 for p, v in zip(patch_size_, is_v))
```

Comment:
```suggestion
        _pad_size = tuple(p if v else 0 for p, v in zip(patch_size_, is_v))
```
Could also have ` _pad_size = tuple(p * v for p, v in zip(patch_size_, is_v))` but that's less clear.